packages/block-library/src/query-loop/index.php

```diff
@@ -36,11 +36,19 @@ function render_block_core_query_loop( $attributes, $content, $block ) {
 
 	$posts      = get_posts( $query );
 	$classnames = '';
+	// `layout` and `align` context come from `Query` block
+	// and is a UI choice to make the handling of the block more
+	// user friendly. The plan is to find a way to make these two
+	// attributes of `QueryLoop`, but keep them in `Query` toolbar
+	// and later absorb the `QueryLoop` toolbar from its parent.
 	if ( isset( $block->context['layout'] ) && isset( $block->context['query'] ) ) {
 		if ( isset( $block->context['layout']['type'] ) && 'flex' === $block->context['layout']['type'] ) {
 			$classnames = "is-flex-container columns-{$block->context['layout']['columns']}";
 		}
 	}
+	if ( isset( $block->context['align'] ) ) {
+		$classnames .= " align{$block->context['align']}";
```

Comment:
Isn't this something that can be applied automatically by the the support flag?

Reply:
Check my comment about the attributes. This is what we had discussed about layout as well, so align goes to Query now.